app/models/legal_aid_application.rb

```diff
@@ -264,6 +264,14 @@ def reset_proceeding_types!
     reset_delegated_functions
   end
 
+  def receives_student_finance?
+    irregular_incomes.map(&:income_type).include? 'student_loan'
+  end
+
+  def value_of_student_finance
+    irregular_incomes.map { |ir| ir.income_type == 'student_loan' ? ir.amount : nil }.first
```

Comment:
I think the responsibility for extracting student loan records from all the other types of irregular income properly lies with the `IrregularIncome` model with a scope

`scope :student_finance, -> { where(income_type: 'student_loan) }`

then these two methods become:

```
def receives_student_finance?
  irregular_incomes.student_finance.any?
end

def value_of_student_finance
  receives_student_finance? ? irregular_incomes.student_student_finance.first.amount : nil
end
```